src/HealthChecks/HealthChecks/src/DependencyInjection/HealthChecksBuilderAddCheckExtensions.cs

```diff
@@ -23,13 +23,15 @@ public static class HealthChecksBuilderAddCheckExtensions
         /// is <c>null</c>, then <see cref="HealthStatus.Unhealthy"/> will be reported.
         /// </param>
         /// <param name="tags">A list of tags that can be used to filter health checks.</param>
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
         /// <returns>The <see cref="IHealthChecksBuilder"/>.</returns>
         public static IHealthChecksBuilder AddCheck(
             this IHealthChecksBuilder builder,
             string name,
             IHealthCheck instance,
             HealthStatus? failureStatus = null,
-            IEnumerable<string> tags = null)
+            IEnumerable<string> tags = null,
+            TimeSpan? timeout = null)
```

Comment:
This is a breaking change also. Here's a short doc on adding overloads to a method with default parameters. https://github.com/dotnet/roslyn/blob/master/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md

Reply:
Backcompat method added